packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -994,5 +1031,19 @@ class _RepeatingSimulation extends Simulation {
   double dx(double timeInSeconds) => (max - min) / _periodInSeconds;
 
   @override
-  bool isDone(double timeInSeconds) => false;
+  bool isDone(double timeInSeconds) {
+    if (repeatCount == null) {
+      // if repeat count is null it is to be consider as repeated simulation
+      return false;
+    }
+
+    // total simulation elapsed time [Fixed to 3 decimal points]
+    final double totalTimeInSeconds = double.parse(
+      (timeInSeconds + _initialT).toStringAsFixed(3),
+    );
+
+    // if [totalTimeInSeconds] elapsed the [exitTimeInSeconds],
+    // consider marking the simulation as "DONE"
+    return totalTimeInSeconds >= _exitTimeInSeconds;
```

Comment:
```suggestion
    return repeatCount == null || timeInSeconds >= _exitTimeInSeconds;
```

Reply:
I just noticed some mistakes in the suggestions I made!

Glad I was able to catch those before you pushed an update 😅

Reply:
Any reason to remove `_initialT` from calculation?

Reply:
hehe, No problem. Thanks for guiding to the better solution.